llvm/utils/TableGen/CallingConvEmitter.cpp

```diff
@@ -35,14 +37,20 @@ class CallingConvEmitter {
   std::map<std::string, std::set<std::string>> DelegateToMap;
 
 public:
-  explicit CallingConvEmitter(const RecordKeeper &R) : Records(R) {}
+  explicit CallingConvEmitter(const RecordKeeper &R) : Records(R), Target(R) {
+    for (const CodeGenRegister &Reg : Target.getRegBank().getRegisters())
+      RegistersByDefName.try_emplace(Reg.getName(), &Reg);
+  }
 
   void run(raw_ostream &O);
 
 private:
   void emitCallingConv(const Record *CC, raw_ostream &O);
   void emitAction(const Record *Action, indent Indent, raw_ostream &O);
   void emitArgRegisterLists(raw_ostream &O);
+
+  StringMap<const CodeGenRegister *> RegistersByDefName;
```

Comment:
`CodeGenTarget` seems to support `getRegisterByName` the difference seems that that name is the asm name vs this is the def name. Would it make sense to change the meaning of `CCAssignToRegTuple` to accept asm names instead of def names and then use this existing function?

Reply:
No, that would not make sense to do-- it would be completely inconsistent with the way names are referenced (record/enum name) in the rest of CallingConv and the back-end as a whole.

Reply:
Makes sense. In that case, if we want to continue using DefNames, would this avoid creating a new map: Use `RecordKeeper::getDef()` to lookup DefName -> Record, and then `CodeGenRegBank::getReg()` for Record -> CodeGenRegister*?

Reply:
No, that `RecordKeeper` lookup will fail for the same reason that it fails in the parser.

Reply:
I looked at this a bit more just for my understanding: It seems "RegisterTuples" are expanded in `CodeGenRegBank` constructor to synthesize the actual register records, and then synthesized records are never added back to `RecordKeeper`, hence the lookup path I suggested will fail. @nvjle is that correct?

Reply:
No, I think its slightly different. I understand that these synthetic records are not available at TableGen parsing time, so there is no way to reference them the standard way. However, once they are synthesized in `CodeGenRegBank` constructor, they could have been added back to the `RecordKeeper` by calling `RecordKeeper::addDef()` and then the 2- step lookup from Name -> Record -> CodeGenRegister * would have worked without needing an additional LUT. But in the absence of that, what this PR is doing seems reasonable.